File: Coding_Platforms/Codewars/Golang/range_extraction.go
package main

import (
	"fmt"
	"strconv"
)

func main() {

	fmt.Println(Solution([]int{1, 2, 3, 6, 7, 8, 9, 10, 11, 20, 22, 23, 31, 33}))
}

func Solution(list []int) string { //range extraction from list

	//create three variables
	//one for result ,one for last_value and one as flag
	result := ""
	var last_value, count int

	for idx, data := range list { //iterate over list

		if idx >= 1 { //if index is greater then one
			if (last_value+1) == data || (last_value-1) == data { //if data of list match with +1 or -1 of previous data then add current data into last_value and count increases by +1 and continue the iteration

				last_value = data
				count += 1
				continue
			}

		}

		if count == 1 { //if count is 1 then add "," in result

			result += ","

		} else if count == 2 { //if count is 2 then add ",", previous data and "," in result

			result += "," + strconv.Itoa(last_value) + ","

		} else if count > 2 { //if count is greater then 2 then add "-", previous data and "," in result

			result += "-" + strconv.Itoa(last_value) + ","
		}

		//assign 1 to count and current data to last_data
		count = 1
		last_value = data

		//add last data into result
		result += strconv.Itoa(last_value)

	}

	// if count is 2 or greater then 2 then add "-" or "," and last value to result because it miss in last value when data in range
	if count == 2 {
		result += "," + strconv.Itoa(last_value)

	} else if count > 2 {
		result += "-" + strconv.Itoa(last_value)
	}

	return result
}

File: Coding_Platforms/Codewars/Python/range_extraction.py
def solution(args): # range extraction

    #create three variables 
    # one for result, one for last value and one for count
    result = ""
    last_value = len(args)-1
    count = 0

    for i,_ in enumerate(args): # iterate over list

        if i<count or i == last_value: # skip iteration  until i != count and if i is last value
            continue

        # assign i to the count 
        # add number to the result
        count = i
        result+= str(args[i])

        while True:# a infinite loop until numbers is continuous

            if count == last_value:# if count is equal to last value then break the look
                break

            elif (count < last_value) & (args[count]+1 == args[count+1]):# if count is less then last value and future value is continuous to current value then add one to count
                count+=1

            else: # if value is not continuous then break the loop
                break

        if count - i > 1: # if count is greater then i then use "-" to result
            result+="-"

        else: # else use "," to result
            result+=","
    
    # add last value to the result
    result += str(args[last_value])
    return result

print(solution([-6,-3,-2,-1,0,1,3,4,5,7,8,9,10,11,14,15,17,18,19,20,22]))
